File: 0671-second-minimum-node-in-a-binary-tree/0671-second-minimum-node-in-a-binary-tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    
    
    public int findSecondMinimumValue(TreeNode root) {
        Queue<TreeNode> queue = new LinkedList<>();
        List<Integer> values = new ArrayList<>();
        
        queue.offer(root);
        while(!queue.isEmpty()) {
            TreeNode cur = queue.poll();
            values.add(cur.val);
            if (cur.left != null) queue.offer(cur.left);
            if (cur.right != null) queue.offer(cur.right);
        }
        
        int max = root.val;
        
        Collections.sort(values);
        for (int num : values) {
            if (max < num) return num;
        }
        
        return -1;
    }
    
}

File: 0671-second-minimum-node-in-a-binary-tree/0671-second-minimum-node-in-a-binary-tree.kt
/**
 * Example:
 * var ti = TreeNode(5)
 * var v = ti.`val`
 * Definition for a binary tree node.
 * class TreeNode(var `val`: Int) {
 *     var left: TreeNode? = null
 *     var right: TreeNode? = null
 * }
 */
class Solution {
   fun findSecondMinimumValue(root: TreeNode?): Int {
        
        if(root == null) return -9
        var elements = arrayListOf<Int>()
        
        fun preOrder(root: TreeNode?){
            if(root == null) return 
            
            preOrder(root.left)
            elements.add(root.`val`)
            preOrder(root.right)
        }
        
        preOrder(root)
        
        elements.sort()
        var ans = elements[0]
        
        for(i in elements){
            if (ans != i ) return i
        }
        
        return -1   
    }
}